### DISCUSSION TITLE
Why don't we have IDisplayObject, IDisplayObjectContainer and IInteractiveObject?

### DISCUSSION THREAD

**Discussion by ylazy** [airsdk/Adobe-Runtime-Support#2781] [Q&A] [answered]
I think we should have something like `IInteractiveObject` extends `IDisplayObjectContainer` extends `IDisplayObject`, then:
`Shape` implement `IDisplayObject`
`Sprite`, `MovieClip`, `Loader` implement `IInteractiveObject`
`DisplayObjectContainer` implement `IDisplayObjectContainer`
`public function addChild(IDisplayObject)` instead of `public function addChild(DisplayObject)`

so we can use:
```as
public interface IComponent extends IInteractiveObject
```
```as
var child:IComponent;
this.addChild(child);
```
Currently this will throw an Error: `Cannot convert IComponent to DisplayObject`

**Comment by ajwfrost** [comment, marked as answer]
Hi

Interesting architectural decision .. if you look at what Adobe did for event dispatcher, there is an interface for that so that classes that need to derive from an object other than EventDispatcher can still act as an event dispatcher.. but I'm not sure what the need is for here?

The one thing I would say is that we can't really make this change. It would break anything that has already been written based on it, which certainly includes Flex, and may also import developers' existing apps...


thanks

**Comment by henke37** [comment]
The thing is that the classes have plenty of native methods and properties that you can't possibly replicate. Only code with direct runtime support can do anything meaningful here.
